content/200-concepts/100-components/01-prisma-schema/06-relations/425-referential-actions/100-cyclic-referential-actions.mdx

```diff
@@ -57,15 +55,22 @@ This will result in the following error:
 Error parsing attribute "@relation": A self-relation must have `onDelete` and `onUpdate` referential actions set to `NoAction` in one of the @relation attributes. Implicit default `onDelete` and `onUpdate` values: `SetNull` and `Cascade`.
 ```
 
-Since the default referential actions (`onDelete: SetNull`, `onUpdate: Cascade`) are creating a cycle, the solution is to explicitly set the `onDelete` and `onUpdate` values to `NoAction` on one of the relations.
+By not defining any actions, Prisma will use the following default values depending if the underlying [scalar fields](../../data-model#scalar-fields) are set to be optional or required.
 
-```prisma highlight=4;delete|5;add
-model Chicken {
-  id        Int      @id @default(autoincrement())
-  cock      Chicken? @relation(name: "a_self_relation")
-  hen       Chicken? @relation(name: "a_self_relation", fields: [chickenId], references: [id])
-  hen       Chicken? @relation(name: "a_self_relation", fields: [chickenId], references: [id], onUpdate: NoAction, onDelete: NoAction)
-  chickenId Int?
+| Clause     | All of the scalar fields are optional | At least one scalar field is required |
+| :--------- | :------------------------------------ | :------------------------------------ |
+| `onDelete` | `SetNull`                             | `NoAction`                            |
+| `onUpdate` | `Cascade`                             | `Cascade`                             |
+
+Since the default referential action for `onUpdate` in the above relation would be `Cascade` and for `onDelete` would be `SetNull`, it creates a cycle and the solution is to explicitly set the `onUpdate` and `onDelete` values to `NoAction`.
```

Comment:
```suggestion
Since the default referential action for `onUpdate` in the above relation would be `Cascade` and for `onDelete` it would be `SetNull`, it creates a cycle and the solution is to explicitly set the `onUpdate` and `onDelete` values to `NoAction`.
```